asim: improve comments and test set up for sma

wenyihu6 opened the pull request:
Epic: CRDB-49117 
Release note: none

---
**asim: move sma and mma to its own folder**

This commit moves sma and mma test data into their own subfolders under testdata
for clearer organization.

---
**asim: add vmodule to tests**

This commit enables vmodule logs by default in asim dd tests.

---
**asim: make RebalancingSnapshotRate configurable**

This commit changes settingsGen.Settings.RebalancingSnapshotRate as a
configurable setting to speed up certain tests.

---
**asim: fix replica state set up string**

Previously, the replica distribution of setup output was inaccurate since it
used target replica and lease count but was not from the actual distribution.
This commit updates the string to reflect the actual distribution.

---
**asim: skip mma tests by default**

This commit skips tests in the mma folder by default since they tend to be
slower, removing the need for skip_under_ci commands.

---
**asim: change store_byte_capacity to *_gib**

This commit changes store_byte_capacity to store_byte_capacity_gib to improve
human readability.

---
**asim: improve assertion output one-off errors**

This commit organizes assertion output to reduce one-off errors - discovered
while testing assertions.

---
**asim: align min key of gen_load and gen_ranges**

Previously, gen_load had a minKey starting at 1, while gen_ranges started at 0.
This commit updates gen_load to start at 0 so both are aligned.

---
**asim: rename multi_store.txt to multi_store_lease.txt**

This commit renames multi_store.txt test to multi_store_lease.txt to reflect
that it asserts on lease balance.

---
**asim: remove load_cluster.txt and zone_config.txt**

This commit remove load_cluster.txt and zone_config.txt tests as they are too
basic and add little value.

---
**asim: fix io_overload**

This test started failing after we fixed the skew distribution, since it relied
on the replica count on store 5 starting and ending at 0. After fixing the skew
distribution, the replica count on store 5 no longer began at 0, even though the
total leases/replica count on s5 didn’t change throughout the test which means
we weren't adding more leases/replicas to the s5. This change updates the test
to force the replica count on store 5 to start at 0 again.

---
**asim: improve comments and test set up for sma**

This commit improves comments and test setup for tests under sma.

wenyihu6 committed:
asim: move sma and mma to its own folder

This commit moves sma and mma test data into their own subfolders under testdata
for clearer organization.

wenyihu6 committed:
asim: fix replica state set up string

Previously, the replica distribution of setup output was inaccurate since it
used target replica and lease count but was not from the actual distribution.
This commit updates the string to reflect the actual distribution.

wenyihu6 committed:
asim: skip mma tests by default

This commit skips tests in the mma folder by default since they tend to be
slower, removing the need for skip_under_ci commands.

wenyihu6 committed:
asim: change store_byte_capacity to *_gib

This commit changes store_byte_capacity to store_byte_capacity_gib to improve
human readability.

wenyihu6 committed:
asim: improve assertion output one-off errors

This commit organizes assertion output to reduce one-off errors - discovered
while testing assertions.

wenyihu6 committed:
asim: align min key of gen_load and gen_ranges

Previously, gen_load had a minKey starting at 1, while gen_ranges started at 0.
This commit updates gen_load to start at 0 so both are aligned.

wenyihu6 committed:
asim: add vmodule to tests

This commit enables vmodule logs by default in asim dd tests.

wenyihu6 committed:
asim: make RebalancingSnapshotRate configurable

This commit changes settingsGen.Settings.RebalancingSnapshotRate as a
configurable setting to speed up certain tests.

wenyihu6 committed:
asim: rename multi_store.txt to multi_store_lease.txt

This commit renames multi_store.txt test to multi_store_lease.txt to reflect
that it asserts on lease balance.

wenyihu6 committed:
asim: remove load_cluster.txt and zone_config.txt

This commit remove load_cluster.txt and zone_config.txt tests as they are too
basic and add little value.

wenyihu6 committed:
asim: improve comments and test set up for sma

This commit improves comments and test setup for tests under sma.

wenyihu6 committed:
asim: fix io_overload

This test started failing after we fixed the skew distribution, since it relied
on the replica count on store 5 starting and ending at 0. After fixing the skew
distribution, the replica count on store 5 no longer began at 0, even though the
total leases/replica count on s5 didn’t change throughout the test which means
we weren't adding more leases/replicas to the s5. This change updates the test
to force the replica count on store 5 to start at 0 again.

tbg reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> nice!

> @tbg reviewed 37 of 37 files at r1, 35 of 35 files at r2, 35 of 35 files at r3, 35 of 35 files at r4, 35 of 35 files at r5, 35 of 35 files at r6, 37 of 37 files at r7, 35 of 35 files at r8, 34 of 34 files at r9, 34 of 34 files at r10, 32 of 32 files at r11, 32 of 32 files at r12, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/154343#-:-OaKZ9HABN5BPtPKQkz8:b-7yytuz)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @wenyihu6)

___
*[`-- commits` line 29 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/154343#-OaKY0Fc4bRRiYCoTWeo:-OaKY0Fc4bRRiYCoTWep:bwcuzlv):*
They tend to be slower because they simulate more complex stuff, or have you noticed any other reasons?
___
*[`pkg/kv/kvserver/asim/tests/datadriven_simulation_test.go` line 742 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/154343#-OaKX8SU3F7sM_Z7aEA1:-OaKX8SU3F7sM_Z7aEA2:byjvk5m) ([raw file](https://github.com/cockroachdb/cockroach/blob/0adbc2b270b7238527f31fe7511c6ae3e07f0b6c/pkg/kv/kvserver/asim/tests/datadriven_simulation_test.go#L742)):*
> ```Go
> }
> 
> func setAllocatorDebugVModule(t *testing.T) (reset func()) {
> ```

I'm okay with this since you probably thought it through, but I'd like to understand

1. does this make the tests slower?
2. what does this provide over the tracing I added recently?


<!-- Sent from Reviewable.io -->

wenyihu6 committed:
asim: place generated output in two subfolders

Previously, we split different test modes into their own subfolders.  This
commit extends the same change to the generated folder, so that its outputs are
also placed into two subfolders. Note that these are only organizational clean
ups - tests were grouped and labeled based on their original purpose. Tests
under sma may still specify cfgs = (mma-only or mma-count) and vice-versa for
tests under mma.

wenyihu6 committed:
Revert "asim: add vmodule to tests"

This reverts commit 0adbc2b270b7238527f31fe7511c6ae3e07f0b6c.

wenyihu6 reviewed (commented):
I pushed another commit to put the generated folder output into two subfolders as well.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/154343#-:-OaKxPsP2h04lio3mdwq:bhryym3)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @tbg)

___
*[`-- commits` line 29 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/154343#-OaKY0Fc4bRRiYCoTWeo:-OaKvvInEDaX61JCS-zs:b-w9ugml):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

They tend to be slower because they simulate more complex stuff, or have you noticed any other reasons?
</blockquote></details>

mma tests usually have longer durations (some run 60–90 minutes, while sma is capped at 20 minutes). And mma tests usually cover more mode cfgs. 

I noticed some of the mma-only tests are actually longer than mma-count for the same test set-up. I’m not certain yet - maybe because mma-only runs first and sets things up, so the simulator is already set up for mma-count, or something else. We can dig into this later when we’re cleaning up the tests tgtr. I added an item in our spreadsheet for this - line 89 in https://docs.google.com/spreadsheets/d/1He-T4QwCdcz-tlfzSRmDHwSAB9FFeIIR4n47nsmtQUw/edit?gid=241535206#gid=241535206.
___
*[`pkg/kv/kvserver/asim/tests/datadriven_simulation_test.go` line 742 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/154343#-OaKX8SU3F7sM_Z7aEA1:-OaKwSGT6V15eKOLsbFP:b-dlox43) ([raw file](https://github.com/cockroachdb/cockroach/blob/0adbc2b270b7238527f31fe7511c6ae3e07f0b6c/pkg/kv/kvserver/asim/tests/datadriven_simulation_test.go#L742)):*
> 1. does this make the tests slower?

You are right - I didn't think it through. I didn't expect the change on more logging to incur a significant more overhead. I ran it on this commit (221.2s) v.s. commit~1 (190.9s). I've pushed a commit to revert this change. 

What’s the reason tracing wouldn’t incur that cost? Is it just because I’m enabling vmodule in places where the tracing wasn’t actually being hit?

> 2. -   what does this provide over the tracing I added recently?

I’ve always run tests with vmodules enabled, so I thought I’d just check this change in. I usually find the full logs helpful for piecing together during investigations, though I didn’t think too deeply about it.


<!-- Sent from Reviewable.io -->
